File: backend/app/api/reports.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session
from typing import List
from app.database import get_db
from app.core.security import get_current_user_id
from app.models.report import Report
from app.schemas.report import ReportResponse, PreviewRequest, PreviewResponse
from app.services.report_service import generate_report

router = APIRouter(prefix="/reports", tags=["reports"])


@router.get("", response_model=List[ReportResponse])
def list_reports(
    limit: int = Query(50, ge=1, le=200),
    db: Session = Depends(get_db),
    user_id: int = Depends(get_current_user_id),
):
    """Giriş yapan kullanıcının geçmiş raporları (en yeniden eskiye)."""
    return (
        db.query(Report)
        .filter(Report.user_id == user_id)
        .order_by(Report.created_at.desc())
        .limit(limit)
        .all()
    )


@router.post("/preview", response_model=PreviewResponse)
def preview_report(
    payload: PreviewRequest,
    user_id: int = Depends(get_current_user_id),
):
    """
    'Şimdi Getir' — bir konu için anlık arama+sıralama yapar, sonuçları döner.
    Veritabanına kayıt YAPMAZ, e-posta GÖNDERMEZ; sadece önizleme amaçlıdır.
    """
    results = generate_report(topic=payload.topic, category=payload.category)
    return PreviewResponse(topic=payload.topic, category=payload.category, results=results)

File: frontend/src/services/reports.ts
import api from './api'
import type { Category } from './subscriptions'

export interface ReportItem {
  title: string
  url: string
  content: string
  score: number | null
  source: string | null
}

export interface Report {
  id: number
  user_id: number
  subscription_id: number | null
  topic: string
  category: string
  results: ReportItem[]
  created_at: string
}

/** Kullanıcının geçmiş raporları (en yeniden eskiye). */
export async function listReports(limit = 50): Promise<Report[]> {
  const { data } = await api.get('/reports', { params: { limit } })
  return data
}

/** 'Şimdi Getir' — bir konu için anlık önizleme (kayıt/e-posta yok). */
export async function previewReport(
  topic: string,
  category: Category | string = 'General',
): Promise<{ topic: string; category: string; results: ReportItem[] }> {
  const { data } = await api.post('/reports/preview', { topic, category })
  return data
}
